core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1078,7 +1078,9 @@ abstract class RDD[T: ClassTag](
       val scale = math.max(math.ceil(math.pow(numPartitions, 1.0 / depth)).toInt, 2)
       // If creating an extra level doesn't help reduce
       // the wall-clock time, we stop tree aggregation.
-      while (numPartitions > scale + numPartitions / scale) {
+
+      // Don't trigger treeAggregation for 5 partitions
+      while (numPartitions > scale + math.ceil(1.0 * numPartitions / scale)) {
```

Comment:
Yeah, @mengxr was your intent to add the condition `numPartitions > 5`?
Also, I don't think you need to change the current condition; it's fine. (BTW `.toDouble` is a better idiom for casting than `1.0 *`)